starknet/test/l1-execution.test.ts

```diff
@@ -0,0 +1,435 @@
+      _authenticators: [starkTxAuthenticator.address],
+      _metadata_URI: [],
+      _dao_URI: [],
+    });
+    await account.execute({
+      contractAddress: space.address,
+      entrypoint: 'initialize',
+      calldata: initializeCalldata,
+    });
+
+    const quorum = 1;
+
+    const MockStarknetMessaging = await ethers.getContractFactory('MockStarknetMessaging', signer);
+    const messageCancellationDelay = 5 * 60; // seconds
+    mockStarknetMessaging = await MockStarknetMessaging.deploy(messageCancellationDelay);
+
+    ({ l1AvatarExecutionStrategy, safe } = await safeWithL1AvatarExecutionStrategySetup(
+      signer,
+      mockStarknetMessaging.address,
+      space.address,
+      ethRelayer.address,
+      quorum,
+    ));
+    // Dumping the Starknet state so it can be loaded at the same point for each test
+    await starknet.devnet.dump('dump.pkl');
+  }, 10000000);
+
+  it('should execute a proposal via the Avatar Execution Strategy connected to a Safe', async function () {
+    // // Resetting Starknet Devnet Timestamp
+    // await starknet.devnet.setTime(Date.now()/1000);
+    await starknet.devnet.restart();
+    await starknet.devnet.load('./dump.pkl');
+    await starknet.devnet.loadL1MessagingContract(eth_network, mockStarknetMessaging.address);
+
+    const proposalTx = {
+      to: signer.address,
+      value: 0,
+      data: '0x11',
+      operation: 0,
+      salt: 1,
+    };
+
+    const abiCoder = new ethers.utils.AbiCoder();
+    const executionHash = ethers.utils.keccak256(
+      abiCoder.encode(
+        ['tuple(address to, uint256 value, bytes data, uint8 operation, uint256 salt)[]'],
+        [[proposalTx]],
+      ),
+    );
+    // Represent the execution hash as a Cairo Uint256
+    const executionHashUint256: Uint256 = uint256.bnToUint256(executionHash);
+
+    const executionPayload = [
+      l1AvatarExecutionStrategy.address,
+      executionHashUint256.low,
+      executionHashUint256.high,
+    ];
+
+    // Propose
+    await account.execute({
+      contractAddress: starkTxAuthenticator.address,
+      entrypoint: 'authenticate_propose',
+      calldata: CallData.compile({
+        space: space.address,
+        author: account.address,
+        executionStrategy: {
+          address: ethRelayer.address,
+          params: executionPayload,
+        },
+        userProposalValidationParams: [],
+        metadataURI: [],
+      }),
+    });
+
+    // Vote
+    await account.execute({
+      contractAddress: starkTxAuthenticator.address,
+      entrypoint: 'authenticate_vote',
+      calldata: CallData.compile({
+        space: space.address,
+        voter: account.address,
+        proposalId: { low: '0x1', high: '0x0' },
+        choice: '0x1',
+        userVotingStrategies: [{ index: '0x0', params: [] }],
+        metadataURI: [],
+      }),
+    });
+
+    // TODO: Advance time so that the maxVotingTimestamp is exceeded
+    await starknet.devnet.increaseTime(10);
+    await sleep(10000);
```

Comment:
why do we need the sleep? guessing it's a forgotten line?

Reply:
we needed to advance the ethereum blockchain time and i didnt know how to advance time on there so just used a sleep to manually do it. Figured it out though: https://github.com/snapshot-labs/sx-starknet/blob/571ec086c5328d63a6da80550874193493ed1b51/starknet/test/l1-execution.test.ts#L15C5-L15C5 

so not needed anymore and removed